From d1dbe0a075400c0505e93ebbcdd9dc07cef225bc Mon Sep 17 00:00:00 2001
From: James Cowgill <james410@cowgill.org.uk>
Date: Thu, 14 May 2015 20:58:38 +0100
Subject: [PATCH 2/5] Remove RPATH settings from plugin CMakeLists

If CMAKE_INSTALL_RPATH is set (like it is in the toplevel CMakeLists.txt),
CMake will automatically handle all this RPATH stuff for you so none of these
extra -Wl,-rpath additions are needed.

[modified to not patch files which don't exist in the debian version]
---
 CMakePlugin/CMakeLists.txt                            |  5 -----
 CallGraph/CMakeLists.txt                              |  6 ------
 CodeDesigner/CMakeLists.txt                           |  5 -----
 CodeFormatter/CMakeLists.txt                          |  5 -----
 CodeLite/CMakeLists.txt                               | 13 ++++---------
 CodeLiteDiff/CMakeLists.txt                           |  5 -----
 ContinuousBuild/CMakeLists.txt                        |  5 -----
 Copyright/CMakeLists.txt                              |  5 -----
 DatabaseExplorer/CMakeLists.txt                       |  4 ----
 Debugger/CMakeLists.txt                               |  6 ------
 ExternalTools/CMakeLists.txt                          |  8 +-------
 Gizmos/CMakeLists.txt                                 |  7 +------
 HelpPlugin/CMakeLists.txt                             |  5 -----
 LLDBDebugger/CMakeLists.txt                           |  5 -----
 LLDBDebugger/codelite-lldb/CMakeLists.txt             |  5 -----
 LiteEditor/CMakeLists.txt                             |  5 -----
 MacBundler/CMakeLists.txt                             |  5 -----
 MemCheck/CMakeLists.txt                               |  7 +------
 Outline/CMakeLists.txt                                |  9 +--------
 Plugin/CMakeLists.txt                                 | 11 +++--------
 QmakePlugin/CMakeLists.txt                            |  5 -----
 Runtime/templates/gizmos/CMakeLists.txt.plugin.wizard |  5 -----
 SFTP/CMakeLists.txt                                   |  6 ------
 SnipWiz/CMakeLists.txt                                |  6 ------
 SpellChecker/CMakeLists.txt                           |  3 ---
 Subversion2/CMakeLists.txt                            |  8 +-------
 Tweaks/CMakeLists.txt                                 |  6 ------
 UnitTestCPP/CMakeLists.txt                            |  6 ------
 WebTools/CMakeLists.txt                               |  5 -----
 WordCompletion/CMakeLists.txt                         |  8 +-------
 ZoomNavigator/CMakeLists.txt                          |  8 +-------
 abbreviation/CMakeLists.txt                           |  5 -----
 cmake/Modules/plugin.cmake                            |  3 ---
 codelite_make/CMakeLists.txt                          |  5 -----
 codelite_terminal/CMakeLists.txt                      |  5 -----
 codelitegcc/CMakeLists.txt                            |  5 -----
 codelitephp/CMakeLists.txt                            |  5 -----
 cppchecker/CMakeLists.txt                             |  5 -----
 cscope/CMakeLists.txt                                 |  6 ------
 git/CMakeLists.txt                                    |  5 -----
 sdk/codelite_indexer/CMakeLists.txt                   |  5 -----
 sdk/databaselayer/CMakeLists.txt                      |  5 -----
 sdk/wxshapeframework/CMakeLists.txt                   |  5 -----
 wxformbuilder/CMakeLists.txt                          |  5 -----
 46 files changed, 15 insertions(+), 253 deletions(-)

diff --git a/CMakePlugin/CMakeLists.txt b/CMakePlugin/CMakeLists.txt
index eddb29d..5bc6424 100644
--- a/CMakePlugin/CMakeLists.txt
+++ b/CMakePlugin/CMakeLists.txt
@@ -38,11 +38,6 @@ if (USE_PCH)
     add_definitions(-Winvalid-pch)
 endif (USE_PCH)
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/CallGraph/CMakeLists.txt b/CallGraph/CMakeLists.txt
index 3dba7fd..8f201b4 100644
--- a/CallGraph/CMakeLists.txt
+++ b/CallGraph/CMakeLists.txt
@@ -26,12 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 if (UNIX AND NOT APPLE)
diff --git a/CodeDesigner/CMakeLists.txt b/CodeDesigner/CMakeLists.txt
index 40679c7..d8f2d85 100644
--- a/CodeDesigner/CMakeLists.txt
+++ b/CodeDesigner/CMakeLists.txt
@@ -21,11 +21,6 @@ add_definitions(-DWXUSINGDLL_WXSQLITE3)
 add_definitions(-DWXUSINGDLL_CL)
 add_definitions(-DWXUSINGDLL_SDK)
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
diff --git a/CodeFormatter/CMakeLists.txt b/CodeFormatter/CMakeLists.txt
index e447950..483bc60 100644
--- a/CodeFormatter/CMakeLists.txt
+++ b/CodeFormatter/CMakeLists.txt
@@ -36,11 +36,6 @@ if ( APPLE )
     add_definitions(-fPIC)
 endif()
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
diff --git a/CodeLite/CMakeLists.txt b/CodeLite/CMakeLists.txt
index 57c743d..20b2cbb 100644
--- a/CodeLite/CMakeLists.txt
+++ b/CodeLite/CMakeLists.txt
@@ -91,16 +91,11 @@ endif()
 add_definitions(-DWXUSINGDLL_WXSQLITE3)
 
 # Add RPATH
-if (NOT MINGW)
-if ( WXC_APP )
+if (NOT MINGW AND WXC_APP)
     string( REPLACE "codelite" "wxcrafter" WXC_LIBS_DIR ${PLUGINS_DIR}) 
-    set (LINKER_OPTIONS -Wl,-rpath,"${WXC_LIBS_DIR}:${PLUGINS_DIR}")
-    message( "-- libcodelite.so is using RPATH set to ${WXC_LIBS_DIR}:${PLUGINS_DIR}")
-else ( WXC_APP )
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-    message( "-- libcodelite.so is using RPATH set to ${PLUGINS_DIR}")
-endif ( WXC_APP )
-endif (NOT MINGW)
+    set (LINKER_OPTIONS -Wl,-rpath,"${WXC_LIBS_DIR}")
+    message( "-- libcodelite.so is using RPATH set to ${WXC_LIBS_DIR}")
+endif ()
 FILE(GLOB SRCS "*.cpp" "../sdk/codelite_indexer/network/*.cpp" "SocketAPI/*.cpp")
 
 # Define the output
diff --git a/CodeLiteDiff/CMakeLists.txt b/CodeLiteDiff/CMakeLists.txt
index 121d2a1..851b550 100644
--- a/CodeLiteDiff/CMakeLists.txt
+++ b/CodeLiteDiff/CMakeLists.txt
@@ -34,11 +34,6 @@ if (USE_PCH)
     add_definitions(-Winvalid-pch)
 endif (USE_PCH)
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/ContinuousBuild/CMakeLists.txt b/ContinuousBuild/CMakeLists.txt
index d37c37a..c4602b8 100644
--- a/ContinuousBuild/CMakeLists.txt
+++ b/ContinuousBuild/CMakeLists.txt
@@ -26,11 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 if (UNIX AND NOT APPLE)
diff --git a/Copyright/CMakeLists.txt b/Copyright/CMakeLists.txt
index 09e1722..a4040b0 100644
--- a/Copyright/CMakeLists.txt
+++ b/Copyright/CMakeLists.txt
@@ -26,11 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 if (UNIX AND NOT APPLE)
diff --git a/DatabaseExplorer/CMakeLists.txt b/DatabaseExplorer/CMakeLists.txt
index b6c8e16..d6955e3 100644
--- a/DatabaseExplorer/CMakeLists.txt
+++ b/DatabaseExplorer/CMakeLists.txt
@@ -75,10 +75,6 @@ add_definitions(-DWXUSINGDLL_WXSQLITE3)
 add_definitions(-DWXUSINGDLL_CL)
 add_definitions(-DWXUSINGDLL_SDK)
 add_definitions(-DDBL_USE_SQLITE)
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
 
 FILE(GLOB SRCS "*.cpp")
 
diff --git a/Debugger/CMakeLists.txt b/Debugger/CMakeLists.txt
index 413ed2e..534df55 100644
--- a/Debugger/CMakeLists.txt
+++ b/Debugger/CMakeLists.txt
@@ -35,12 +35,6 @@ if ( APPLE )
     add_definitions(-fPIC)
 endif()
 
-
-# Add RPATH
-if (UNIX)
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
diff --git a/ExternalTools/CMakeLists.txt b/ExternalTools/CMakeLists.txt
index 1071308..f4d5bfc 100644
--- a/ExternalTools/CMakeLists.txt
+++ b/ExternalTools/CMakeLists.txt
@@ -35,12 +35,6 @@ if ( APPLE )
     add_definitions(-fPIC)
 endif()
 
-
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
@@ -56,4 +50,4 @@ target_link_libraries(${PLUGIN_NAME}
                       plugin
                       )
 # Installation destination
-CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
+CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/Gizmos/CMakeLists.txt b/Gizmos/CMakeLists.txt
index 603054e..29f3f18 100644
--- a/Gizmos/CMakeLists.txt
+++ b/Gizmos/CMakeLists.txt
@@ -26,11 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
@@ -54,4 +49,4 @@ target_link_libraries(${PLUGIN_NAME}
                       plugin
                       )
 # Installation destination
-CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
+CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/HelpPlugin/CMakeLists.txt b/HelpPlugin/CMakeLists.txt
index 4788085..0ca9f48 100644
--- a/HelpPlugin/CMakeLists.txt
+++ b/HelpPlugin/CMakeLists.txt
@@ -40,11 +40,6 @@ if ( APPLE )
     add_definitions(-fPIC)
 endif()
 
-# Add RPATH
-if (UNIX)
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
diff --git a/LLDBDebugger/CMakeLists.txt b/LLDBDebugger/CMakeLists.txt
index 25cfb47..4f283e9 100644
--- a/LLDBDebugger/CMakeLists.txt
+++ b/LLDBDebugger/CMakeLists.txt
@@ -71,11 +71,6 @@ if (WITH_LLDB MATCHES 1)
         add_definitions(-DWXUSINGDLL_CL)
         add_definitions(-DWXUSINGDLL_SDK)
 
-        # Add RPATH
-        if (UNIX)
-        set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-        endif (UNIX)
-
         ## By default, use the sources under the current folder
         FILE(GLOB PLUGIN_SRCS "*.cpp")
 
diff --git a/LLDBDebugger/codelite-lldb/CMakeLists.txt b/LLDBDebugger/codelite-lldb/CMakeLists.txt
index 5c5c906..10cab9b 100644
--- a/LLDBDebugger/codelite-lldb/CMakeLists.txt
+++ b/LLDBDebugger/codelite-lldb/CMakeLists.txt
@@ -8,11 +8,6 @@ else ( APPLE )
     find_package(wxWidgets COMPONENTS std aui propgrid stc richtext ribbon REQUIRED)
 endif ( APPLE )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRC "*.cpp")
 include_directories(${CL_SRC_ROOT}/LLDBDebugger)
 
diff --git a/LiteEditor/CMakeLists.txt b/LiteEditor/CMakeLists.txt
index 92f7f57..b1e401e 100644
--- a/LiteEditor/CMakeLists.txt
+++ b/LiteEditor/CMakeLists.txt
@@ -62,11 +62,6 @@ if ( USE_CLANG )
   include_directories(${CLANG_INCLUDE})
 endif( USE_CLANG )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 set (RES_FILE "")
diff --git a/MacBundler/CMakeLists.txt b/MacBundler/CMakeLists.txt
index 17fcd41..55c0bc9 100644
--- a/MacBundler/CMakeLists.txt
+++ b/MacBundler/CMakeLists.txt
@@ -30,11 +30,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
diff --git a/MemCheck/CMakeLists.txt b/MemCheck/CMakeLists.txt
index e5723ee..3d2f53c 100644
--- a/MemCheck/CMakeLists.txt
+++ b/MemCheck/CMakeLists.txt
@@ -43,11 +43,6 @@ if(UNIX AND NOT APPLE)
         add_definitions(-fPIC)
     endif()
 
-
-
-    # Add RPATH
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-
     FILE(GLOB SRCS "*.cpp")
 
     # Define the output
@@ -61,4 +56,4 @@ if(UNIX AND NOT APPLE)
     add_dependencies(${PLUGIN_NAME} plugin)
     install(TARGETS ${PLUGIN_NAME} DESTINATION ${PLUGINS_DIR})
 
-endif()
\ No newline at end of file
+endif()
diff --git a/Outline/CMakeLists.txt b/Outline/CMakeLists.txt
index 2483bb5..71145d6 100644
--- a/Outline/CMakeLists.txt
+++ b/Outline/CMakeLists.txt
@@ -36,13 +36,6 @@ if ( APPLE )
     add_definitions(-fPIC)
 endif()
 
-
-
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
@@ -57,4 +50,4 @@ target_link_libraries(${PLUGIN_NAME}
                       plugin
                       )
 # Installation destination
-CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
+CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/Plugin/CMakeLists.txt b/Plugin/CMakeLists.txt
index ade4992..928c480 100644
--- a/Plugin/CMakeLists.txt
+++ b/Plugin/CMakeLists.txt
@@ -77,15 +77,10 @@ add_definitions(-DWXUSINGDLL_WXSQLITE3)
 add_definitions(-DWXUSINGDLL_CL)
 
 # Add RPATH
-if(NOT MINGW)
-if ( WXC_APP )
+if(NOT MINGW AND WXC_APP)
     string( REPLACE "codelite" "wxcrafter" WXC_LIBS_DIR ${PLUGINS_DIR})
-    set (LINKER_OPTIONS -Wl,-rpath,"${WXC_LIBS_DIR}:${PLUGINS_DIR}")
-    message( "-- libplugin.so is using RPATH set to ${WXC_LIBS_DIR}:${PLUGINS_DIR}")
-else ( WXC_APP )
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-    message( "-- libplugin.so is using RPATH set to ${PLUGINS_DIR}")
-endif ( WXC_APP )
+    set (LINKER_OPTIONS -Wl,-rpath,"${WXC_LIBS_DIR}")
+    message( "-- libplugin.so is using RPATH set to ${WXC_LIBS_DIR}")
 endif()
 
 FILE(GLOB SRCS "*.cpp")
diff --git a/QmakePlugin/CMakeLists.txt b/QmakePlugin/CMakeLists.txt
index 2187bfc..e6ecddd 100644
--- a/QmakePlugin/CMakeLists.txt
+++ b/QmakePlugin/CMakeLists.txt
@@ -26,11 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/Runtime/templates/gizmos/CMakeLists.txt.plugin.wizard b/Runtime/templates/gizmos/CMakeLists.txt.plugin.wizard
index 15b7447..bbc49e8 100644
--- a/Runtime/templates/gizmos/CMakeLists.txt.plugin.wizard
+++ b/Runtime/templates/gizmos/CMakeLists.txt.plugin.wizard
@@ -40,11 +40,6 @@ if ( APPLE )
     add_definitions(-fPIC)
 endif()
 
-# Add RPATH
-if (UNIX)
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
diff --git a/SFTP/CMakeLists.txt b/SFTP/CMakeLists.txt
index 79d5f9e..51b80e5 100644
--- a/SFTP/CMakeLists.txt
+++ b/SFTP/CMakeLists.txt
@@ -26,12 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/SnipWiz/CMakeLists.txt b/SnipWiz/CMakeLists.txt
index 2d82f0c..18d8db4 100644
--- a/SnipWiz/CMakeLists.txt
+++ b/SnipWiz/CMakeLists.txt
@@ -26,12 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/SpellChecker/CMakeLists.txt b/SpellChecker/CMakeLists.txt
index abc9dbf..aa8978d 100644
--- a/SpellChecker/CMakeLists.txt
+++ b/SpellChecker/CMakeLists.txt
@@ -67,9 +67,6 @@ else ()
         add_definitions(-Winvalid-pch)
     endif ( USE_PCH )
 
-    # Add RPATH
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-
     FILE(GLOB SRCS "*.cpp")
 
     # Define the output
diff --git a/Subversion2/CMakeLists.txt b/Subversion2/CMakeLists.txt
index b8c461f..c8ee97d 100644
--- a/Subversion2/CMakeLists.txt
+++ b/Subversion2/CMakeLists.txt
@@ -36,12 +36,6 @@ if ( APPLE )
     add_definitions(-fPIC)
 endif()
 
-
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
@@ -57,4 +51,4 @@ target_link_libraries(${PLUGIN_NAME}
                       )
 
 # Installation destination
-CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
+CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/Tweaks/CMakeLists.txt b/Tweaks/CMakeLists.txt
index 8ceb94c..3f1922b 100644
--- a/Tweaks/CMakeLists.txt
+++ b/Tweaks/CMakeLists.txt
@@ -26,12 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/UnitTestCPP/CMakeLists.txt b/UnitTestCPP/CMakeLists.txt
index cc004d7..fc7786c 100644
--- a/UnitTestCPP/CMakeLists.txt
+++ b/UnitTestCPP/CMakeLists.txt
@@ -26,12 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/WebTools/CMakeLists.txt b/WebTools/CMakeLists.txt
index 6510750..51bc546 100644
--- a/WebTools/CMakeLists.txt
+++ b/WebTools/CMakeLists.txt
@@ -30,11 +30,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/WordCompletion/CMakeLists.txt b/WordCompletion/CMakeLists.txt
index a397c81..6129e30 100644
--- a/WordCompletion/CMakeLists.txt
+++ b/WordCompletion/CMakeLists.txt
@@ -26,12 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
@@ -54,4 +48,4 @@ target_link_libraries(${PLUGIN_NAME}
                       libcodelite
                       plugin
                       )
-CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
+CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/ZoomNavigator/CMakeLists.txt b/ZoomNavigator/CMakeLists.txt
index 5779819..cc54d61 100644
--- a/ZoomNavigator/CMakeLists.txt
+++ b/ZoomNavigator/CMakeLists.txt
@@ -26,12 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
@@ -55,4 +49,4 @@ target_link_libraries(${PLUGIN_NAME}
                       libcodelite
                       plugin
                       )
-CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
+CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/abbreviation/CMakeLists.txt b/abbreviation/CMakeLists.txt
index d472cf0..e4991fc 100644
--- a/abbreviation/CMakeLists.txt
+++ b/abbreviation/CMakeLists.txt
@@ -26,11 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/cmake/Modules/plugin.cmake b/cmake/Modules/plugin.cmake
index f2788ba..5b75878 100644
--- a/cmake/Modules/plugin.cmake
+++ b/cmake/Modules/plugin.cmake
@@ -50,9 +50,6 @@ function( CL_PLUGIN PLUGIN_NAME )
         add_definitions(-include "${CL_PCH_FILE}")
         add_definitions(-Winvalid-pch)
     endif ( USE_PCH )
-
-    # Add RPATH
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
     
     ## By default, use the sources under the current folder
     FILE(GLOB_RECURSE PLUGIN_SRCS "${CMAKE_CURRENT_LIST_DIR}/*.cpp" "${CMAKE_CURRENT_LIST_DIR}/*.c")
diff --git a/codelite_make/CMakeLists.txt b/codelite_make/CMakeLists.txt
index f75a6e1..20174cf 100644
--- a/codelite_make/CMakeLists.txt
+++ b/codelite_make/CMakeLists.txt
@@ -14,11 +14,6 @@ find_package(wxWidgets COMPONENTS core base REQUIRED)
 # Include paths
 #include_directories("${CL_SRC_ROOT}/Plugin" "${CL_SRC_ROOT}/sdk/wxsqlite3/include" "${CL_SRC_ROOT}/CodeLite" "${CL_SRC_ROOT}/PCH" "${CL_SRC_ROOT}/Interfaces")
 
-# Add RPATH
-if ( UNIX )
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif ( UNIX )
-
 # we need wxWidgets flags to be set only for the c++ files, so we do it like this
 # by setting the CMAKE_CXX_FLAGS
 if ( NOT MINGW )
diff --git a/codelite_terminal/CMakeLists.txt b/codelite_terminal/CMakeLists.txt
index 9053477..e0aa46e 100644
--- a/codelite_terminal/CMakeLists.txt
+++ b/codelite_terminal/CMakeLists.txt
@@ -11,11 +11,6 @@ else ( UNIX AND NOT APPLE )
     find_package(wxWidgets REQUIRED)
 endif ( UNIX OR MINGW AND NOT APPLE )
 
-# Add RPATH
-if ( UNIX )
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif ( UNIX )
-
 # we need wxWidgets flags to be set only for the c++ files, so we do it like this
 # by setting the CMAKE_CXX_FLAGS
 if ( NOT MINGW )
diff --git a/codelitegcc/CMakeLists.txt b/codelitegcc/CMakeLists.txt
index 298cec1..f55faac 100644
--- a/codelitegcc/CMakeLists.txt
+++ b/codelitegcc/CMakeLists.txt
@@ -5,11 +5,6 @@ cmake_minimum_required(VERSION 2.6.2)
 # visual studio, and in our makefiles. 
 project(codelite-cc)
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
diff --git a/codelitephp/CMakeLists.txt b/codelitephp/CMakeLists.txt
index ed4e416..d6ccf82 100644
--- a/codelitephp/CMakeLists.txt
+++ b/codelitephp/CMakeLists.txt
@@ -45,11 +45,6 @@ add_definitions(-DWXUSINGDLL_WXSQLITE3)
 add_definitions(-DWXUSINGDLL_CL)
 add_definitions(-DWXUSINGDLL_SDK)
 
-if (UNIX)
-# Add RPATH
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif()
-
 if(CMAKE_BUILD_TYPE MATCHES Debug OR CMAKE_BUILD_TYPE MATCHES DebugFull)
     ## Debug build of codelite
     set( CL_LIB_DIR lib)
diff --git a/cppchecker/CMakeLists.txt b/cppchecker/CMakeLists.txt
index 2d6e265..10d9048 100644
--- a/cppchecker/CMakeLists.txt
+++ b/cppchecker/CMakeLists.txt
@@ -21,11 +21,6 @@ add_definitions(-DWXUSINGDLL_WXSQLITE3)
 add_definitions(-DWXUSINGDLL_CL)
 add_definitions(-DWXUSINGDLL_SDK)
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 if (UNIX AND NOT APPLE)
diff --git a/cscope/CMakeLists.txt b/cscope/CMakeLists.txt
index 847b0c1..94a7c71 100644
--- a/cscope/CMakeLists.txt
+++ b/cscope/CMakeLists.txt
@@ -35,12 +35,6 @@ if ( APPLE )
     add_definitions(-fPIC)
 endif()
 
-
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 FILE(GLOB SRCS "*.cpp")
 
 # Define the output
diff --git a/git/CMakeLists.txt b/git/CMakeLists.txt
index 660f7b6..4bb6079 100644
--- a/git/CMakeLists.txt
+++ b/git/CMakeLists.txt
@@ -21,11 +21,6 @@ add_definitions(-DWXUSINGDLL_WXSQLITE3)
 add_definitions(-DWXUSINGDLL_CL)
 add_definitions(-DWXUSINGDLL_SDK)
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/sdk/codelite_indexer/CMakeLists.txt b/sdk/codelite_indexer/CMakeLists.txt
index 55bcaed..b6d5e3c 100644
--- a/sdk/codelite_indexer/CMakeLists.txt
+++ b/sdk/codelite_indexer/CMakeLists.txt
@@ -14,11 +14,6 @@ find_package(wxWidgets COMPONENTS core base REQUIRED)
 # Include paths
 #include_directories("${CL_SRC_ROOT}/Plugin" "${CL_SRC_ROOT}/sdk/wxsqlite3/include" "${CL_SRC_ROOT}/CodeLite" "${CL_SRC_ROOT}/PCH" "${CL_SRC_ROOT}/Interfaces")
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 # we need wxWidgets flags to be set only for the c++ files, so we do it like this
 # by setting the CMAKE_CXX_FLAGS
 if ( NOT MINGW )
diff --git a/sdk/databaselayer/CMakeLists.txt b/sdk/databaselayer/CMakeLists.txt
index ea8556b..407d035 100644
--- a/sdk/databaselayer/CMakeLists.txt
+++ b/sdk/databaselayer/CMakeLists.txt
@@ -23,11 +23,6 @@ if(WIN32)
     add_definitions(-DWXMAKINGDLL_DATABASELAYER)
 endif(WIN32)
 
-# Add RPATH
-if (UNIX)
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/sdk/wxshapeframework/CMakeLists.txt b/sdk/wxshapeframework/CMakeLists.txt
index 90c4059..308cec4 100644
--- a/sdk/wxshapeframework/CMakeLists.txt
+++ b/sdk/wxshapeframework/CMakeLists.txt
@@ -24,11 +24,6 @@ endif ( USE_PCH )
 
 include_directories("${CL_SRC_ROOT}/sdk/wxshapeframework/include" "${CL_SRC_ROOT}/sdk/wxshapeframework/src")
 
-# Add RPATH
-if (UNIX)
-    set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
diff --git a/wxformbuilder/CMakeLists.txt b/wxformbuilder/CMakeLists.txt
index 4036482..a61edc1 100644
--- a/wxformbuilder/CMakeLists.txt
+++ b/wxformbuilder/CMakeLists.txt
@@ -26,11 +26,6 @@ if ( USE_PCH )
     add_definitions(-Winvalid-pch)
 endif ( USE_PCH )
 
-# Add RPATH
-if (UNIX)
-set (LINKER_OPTIONS -Wl,-rpath,"${PLUGINS_DIR}")
-endif (UNIX)
-
 if (UNIX AND NOT APPLE)
     set ( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC" )
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC" )
-- 
2.1.4

